crates/rpc/src/cairo/ext_py.rs

```diff
@@ -879,38 +874,40 @@ mod tests {
             .unwrap();
         let class_commitment = ClassCommitment(Felt::ZERO);
 
+        let block = StarknetBlock {
+            number: BlockNumber::new_or_panic(1),
+            hash: BlockHash(felt_bytes!(b"some blockhash somewhere")),
+            state_commmitment: StateCommitment::calculate(storage_commitment, class_commitment),
+            timestamp: BlockTimestamp::new_or_panic(1),
+            gas_price: GasPrice(1),
+            sequencer_address: SequencerAddress(Felt::ZERO),
+            transaction_commitment: None,
+            event_commitment: None,
+        };
+
         // create a block with the global root
         StarknetBlocksTable::insert(
             tx,
-            &StarknetBlock {
-                number: BlockNumber::new_or_panic(1),
-                hash: BlockHash(felt_bytes!(b"some blockhash somewhere")),
-                state_commmitment: StateCommitment::calculate(storage_commitment, class_commitment),
-                timestamp: BlockTimestamp::new_or_panic(1),
-                gas_price: GasPrice(1),
-                sequencer_address: SequencerAddress(Felt::ZERO),
-                transaction_commitment: None,
-                event_commitment: None,
-            },
+            &block,
             &StarknetVersion::default(),
             storage_commitment,
             class_commitment,
         )
         .unwrap();
 
-        if false {
-            let mut stmt = tx
-                .prepare("select starknet_block_hash from global_state")
-                .unwrap();
-            let mut rows = stmt.query([]).unwrap();
-            while let Some(row) = rows.next().unwrap() {
-                let first = row.get_ref(0).expect("get column");
+        CanonicalBlocksTable::insert(tx, block.number, block.hash).unwrap();
 
-                println!("{first:?}");
+        ContractUpdatesTable::insert(
+            tx,
+            block.number,
+            test_contract_address,
+            test_contract_class_hash,
+        )
+        .unwrap();
 
-                let first = first.as_blob().expect("cannot read it as a blob");
-                println!("{}", hex::encode(first));
-            }
+        for (address, value) in storage_updates {
+            StorageUpdatesTable::insert(tx, block.number, test_contract_address, address, value)
+                .unwrap();
```

Comment:
Should we consider using `insert_canonical_state_diff` instead? And not having `StorageUpdatesTable` and `ContractUpdatesTable` at all.

I'm hoping we can reduce the surface API of storage - in particular to just the API from the pov of sync, something like:

1. insert state diff
2. insert block header + body
3. insert class definition

Which will also mean less churn in test code (hopefully) if we change table structure etc. Of course the rest of storage doesn't look like this, but maybe its something to work towards, wdyt?

Reply:
Yeah, that's totally possible. I've updated the code to construct a `StateDiff` instead and call `insert_canonical_state_diff()`.